### PR TITLE
Add explicit support for python 3.12

### LINKED ISSUE #1425: Add support for python 3.12
python 3.12 was released 10/2/2023.

### PR DESCRIPTION
**Pull request by phargogh**
Fixes #1425

## Checklist
- [ ] Updated HISTORY.rst and link to any relevant issue (if these changes are user-facing)
~~- [ ] Updated the user's guide (if needed)~~
~~- [ ] Tested the Workbench UI (if relevant)~~

### CODE REVIEW

**Comment by phargogh** [comment]
The binary builds are currently failing due to #1553, patched in #1555.  Waiting until #1555 is merged to merge it in here.

**Review by emlys — APPROVED** [review]
👍

File: requirements.txt
Comment (emlys) [inline requirements.txt:13]: Is this redundant to the constraints file?

Comment (phargogh) [reply requirements.txt]: Sure is!  Patched in 784735bcf

**Pull request merged**
